src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -106,6 +112,92 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
 
   const isEditing = !!artwork?.internalID
 
+  const backModal = (
+    <ModalDialog
+      title="Leave without saving?"
+      onClose={() => setShouldShowBackModal(false)}
+      width={["100%", 600]}
+      footer={
+        <>
+          <Button
+            // @ts-ignore
+            as={RouterLink}
+            to={
+              isEditing
+                ? `/my-collection/artwork/${artwork.internalID}`
+                : "/my-collection"
+            }
+            width="100%"
+            data-testid="leave-button"
+          >
+            Leave Without Saving
+          </Button>
+          <Button
+            onClick={() => setShouldShowBackModal(false)}
+            variant="secondaryNeutral"
+            mt={2}
+            width="100%"
+          >
+            {isEditing ? "Continue Editing" : "Continue Uploading Artwork"}
+          </Button>
+        </>
+      }
+    >
+      <Text>
+        {isEditing
+          ? "Changes you have made so far will not be saved."
+          : "Your artwork will not be added to My Collection."}
+      </Text>
+    </ModalDialog>
+  )
+
+  const handleDelete = async () => {
+    try {
+      await deleteArtworkRequest({
+        artworkId: artwork?.internalID!,
+      }).then(() => router.push({ pathname: "/settings/my-collection" }))
+    } catch (error) {
+      logger.error(`Artwork not deleted`, error)
+
+      sendToast({
+        variant: "error",
+        message: "An error occurred",
+        description: "Please contact support@artsymail.com",
+      })
+
+      return
+    }
+  }
+
+  const deletionModal = (
```

Comment:
What do you think about moving both Modals into their own files? This file is getting pretty long....

Reply:
Agree :)